link.x

```diff
@@ -17,6 +17,14 @@ SECTIONS
     _einterrupts = .;
   } > FLASH
 
+  PROVIDE(_stext = _einterrupts);
+  _reserved_size = _stext - _einterrupts;
+
+  .reserved : ALIGN(4)
+  {
+    . += _reserved_size;
```

Comment:
nit: could you change the RHS to `(_stext - _einterrupts)` and remove the `_reserved_size` symbol / variable from this file? One can bind to all the symbols / variables used in this file so it's better to keep their numbers small to reduce the chance of misuse.

``` rust
extern {
    static _reserved_size: u32;
}
let address = unsafe { &_reserved_size as *const _ as usize };
// ...
```